examples/tiny-cluster-hpa.yaml

```diff
@@ -0,0 +1,193 @@
+        druid.sql.enable=false
+      extra.jvm.options: |-
+        -Xmx1G
+        -Xms1G
+      volumeMounts:
+        - mountPath: /druid/data
+          name: data-volume
+      volumes:
+        - name: data-volume
+          emptyDir: {}
+      resources:
+        requests:
+          memory: "2G"
+          cpu: "2"
+        limits:
+          memory: "2G"
+          cpu: "2"
+      autoscaler:
+        maxReplicas: 10
+        minReplicas: 3
+        scaleTargetRef:
+           apiVersion: apps/v1
+           kind: StatefulSet
+           name: druid-tiny-cluster-brokers
+        metrics:
+         - type: Resource
+           resource:
+            name: cpu
+            targetAverageUtilization: 50
+      
+    coordinators:
+      nodeType: "coordinator"
+      druid.port: 8088
+      nodeConfigMountPath: "/opt/druid/conf/druid/cluster/master/coordinator-overlord"
+      runtime.properties: |
+        druid.service=druid/coordinator
+
+        # HTTP server threads
+        druid.coordinator.startDelay=PT30S
+        druid.coordinator.period=PT30S
+
+        # Configure this coordinator to also run as Overlord
+        druid.coordinator.asOverlord.enable=true
+        druid.coordinator.asOverlord.overlordService=druid/overlord
+        druid.indexer.queue.startDelay=PT30S
+        druid.indexer.runner.type=local
+      extra.jvm.options: |-
+        -Xmx1G
+        -Xms1G
+      volumeMounts:
+        - mountPath: /druid/data
+          name: data-volume
+      volumes:
+        - name: data-volume
+          emptyDir: {}
+      resources:
+        requests:
+          memory: "2G"
+          cpu: "2"
+        limits:
+          memory: "2G"
+          cpu: "2"
+      autoscaler:
+        maxReplicas: 10
+        minReplicas: 1
+        scaleTargetRef:
+           apiVersion: apps/v1
+           kind: StatefulSet
+           name: druid-tiny-cluster-coordinators
+        metrics:
+         - type: Resource
+           resource:
+            name: cpu
+            targetAverageUtilization: 60
+            
+    historicals:
+      nodeType: "historical"
+      druid.port: 8088
+      nodeConfigMountPath: "/opt/druid/conf/druid/cluster/data/historical"
+      runtime.properties: |
+        druid.service=druid/historical
+        druid.server.http.numThreads=5
+        druid.processing.buffer.sizeBytes=1
+        druid.processing.numMergeBuffers=1
+        druid.processing.numThreads=1
+        # Segment storage
+        druid.segmentCache.locations=[{\"path\":\"/druid/data/segments\",\"maxSize\":10737418240}]
+        druid.server.maxSize=10737418240
+      extra.jvm.options: |-
+        -Xmx1G
+        -Xms1G
+      volumeMounts:
+        - mountPath: /druid/data
+          name: data-volume
+      volumes:
+        - name: data-volume
+          emptyDir: {}
+      resources:
+        requests:
+          memory: "2G"
+          cpu: "2"
+        limits:
+          memory: "2G"
+          cpu: "2"
+      autoscaler:
```

Comment:
I don't really think  that autoscaling historical nodes would be a good idea in production due to  coordinator constantly getting confused with segment management, having this in the example might make users believe that it is  a good practice  .

Reply:
valid agreed

Reply:
@himanshug will VPA scaler work for historicals ? For middle managers HPA seems fine. Can you suggest which is suitable for historicals ?

Reply:
Done Changes